interfaces/builtin/desktop_session_control.go

```diff
@@ -0,0 +1,95 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const desktopSessionControlSummary = `allows setting up user processes for a desktop session`
+
+const desktopSessionBaseDeclarationSlots = `
+  desktop-session-control:
+    allow-installation:
+      slot-snap-type:
+        - app
+        - core
+    deny-auto-connection: true
+`
+
+const desktopSessionControlConnectedPlugAppArmor = `
+# Description: Can control the processes in desktop sessions, launch them, etc.
+
+/var/lib/snapd/desktop/applications/{,**} r,
```

Comment:
I think this is already better covered by the `desktop-launch` interface. You'd want to plug that anyway to actually make use of the desktop files, so this can probably go.

Reply:
Likewise this was containing rules to talk to locale1, I removed them as well to use `locale-control` instead. Both `desktop-launch` and `locale-control` were not available on core though, so I added commits to address this.